README.md

```diff
@@ -136,95 +139,146 @@ errors:
     type: MovieId
 ```
 
-### Generating an SDK
+### Starting from OpenAPI
 
-To generate SDKs, you can log in with GitHub from the CLI:
+If you have an existing OpenAPI definition, you can use that as your starting point by specifying the `--openapi` option:
 
-```bash
-fern login
+```
+fern init --openapi ./path/to/openapi.yml
 ```
 
-You can add [generators](compiler/generators) using `fern add`. By default, this
-will publish your SDK to the Fern npm registry (npm.buildwithfern.com).
+The path to the definition can either be a local file, or a HTTP resource. If you start from an OpenAPI definition, the `init` command will forego generating the default project files such as `imbd.yml`, and instead will use your definition as a starting point.
 
-```bash
-fern add fern-typescript-sdk
+For example, you can use the following command to start a new Fern workspace based on an OpenAPI Pet Store definition:
+
+```
+fern init --openapi fern init --openapi https://petstore.swagger.io/v2/swagger.json
 ```
 
-To generate the TypeScript SDK, run:
+Which will allow you to directly generate clients in the next step.
+
+## Generating an SDK
+
+All SDKs are generated server side, so you don't have to worry about installing all the prerequisite tools locally. Before you can start generating files, you'll need to log in:
 
-```bash
+```
+fern login
+```
+
+You'll be prompted to log in to your GitHub account using your preferred browser.
+
+After you have logged in, simply run the generate command to generate your SDKs:
+
+```
 fern generate
 ```
 
-And voila! You just built and published a TypeScript SDK.
+After your code is generated, the resulting artifacts are copied to the output directory specified in `generators.yml` , which we will touch on in the next section.
 
-**Next step:** define _your_ API in Fern. Check out our [docs](https://www.buildwithfern.com/docs/definition) to learn more.
+It is possible to generate the code on your own machine by using the `--local` flag. This does require you to run Docker on your local machine, and Fern will schedule containers to do the code generation automatically. This is further specified in the CLI reference below.
 
-# CLI reference
+## Adding Generators
 
-### `fern generate [--group <group>] [--version <version>]`
+You can add generators using the `fern add`. You can have multiple generators for a single project, e.g. a single `fern generate` call could generate SDKs in Python, TypeScript and Java all in one go.
 
-`fern generate` runs the compiler. It will validate your API and run the
-generators specified in `generators.yml`.
+For instance, if you'd like to add a TypeScript generator to your project, simply run:
 
-If you don't have a `default-group` specified in `generators.yml`, then you must
-specify a group using the `--group` option.
+```
+fern add fern-typescript-sdk
```

Comment:
lets do `fern add fern-typescript-node-sdk`, i wonder if this also fixes the petstore issue you were running into

Reply:
Fixed - I think it was already using that generator though when it failed: 

```
default-group: local
groups:
  local:
    generators:
      - name: fernapi/fern-typescript-node-sdk
        version: 0.7.1
        output:
          location: local-file-system
          path: ../../generated/typescript
      - name: fernapi/fern-openapi
        version: 0.0.28
        output:
          location: local-file-system
          path: ../../generated/openapi

```